crates/steel-core/src/parser/expander.rs

```diff
@@ -625,251 +662,235 @@ impl MacroPattern {
-                        //             ty: TokenType::Quote | TokenType::QuoteTick,
-                        //             ..
-                        //         },
-                        // }) => {
-                        //     println!("MATCHING ON QUOTE");
-                        //     continue;
-                        // }
 
-                        // ExprKind::Atom(Atom {
-                        //     syn:
-                        //         SyntaxObject {
-                        //             ty: TokenType::Identifier(s),
-                        //             ..
-                        //         },
-                        // }) if s.resolve() == "QUOTE" => {
-                        //     println!("MATCHING ON QUOTE");
-                        //     continue;
-                        // }
-                        // ExprKind::Quote()
-                        _ => return false,
+                continue;
+            }
+            MacroPattern::ManyNested(subpatterns) => {
+                for _ in 0..expected_many_captures {
+                    let Some(ExprKind::List(l)) = exprs_iter.next() else {
+                        return false;
+                    };
+
+                    if !match_vec_pattern(subpatterns, &l.args, false) {
+                        return false;
                     }
                 }
-                // Now that we have ManyNested - need to figure out
-                // the recursive step better here
-                MacroPattern::Nested(vec) => {
-                    if let ExprKind::List(l) = val {
-                        // TODO more elegant let* case
-                        if vec.is_empty() && !l.is_empty() {
-                            return false;
-                        }
 
-                        // This solves the destructuring test case
-                        if vec.len() < l.len() && !vec.iter().any(|x| x.is_many()) {
-                            // log::debug!("Matching failed - ellipses doesn't match up");
-                            return false;
-                        }
+                continue;
+            }
+            _ => {}
+        }
 
-                        // Make the recursive call on the next layer
-                        if match_vec_pattern(vec, l) {
-                            continue;
-                        } else {
-                            // log::debug!("Matching failed due to child not matching");
-                            return false;
-                        }
-                    } else if let ExprKind::Quote(_) = val {
-                        // TODO: Come back here
-                        return matches!(vec.as_slice(), &[MacroPattern::Quote(_)]);
-                        // return true;
-                    } else {
-                        // log::debug!("Matching failed - atom does not match list");
+        let Some(val) = exprs_iter.next() else {
+            return false;
+        };
```

Comment:
:information_source: Since we handle `Many*` separately, we can now demand `val` to be present.